src/ucp/core/ucp_rkey.c

```diff
@@ -264,11 +266,13 @@ ucs_status_t ucp_ep_rkey_unpack(ucp_ep_h ep, const void *rkey_buffer,
 
     ucp_rkey_resolve_inner(rkey, ep);
     *rkey_p = rkey;
-    return UCS_OK;
+    status = UCS_OK;
```

Comment:
q: maybe better rewrite it as follows:
```
    ...
    ucp_rkey_resolve_inner(rkey, ep);
    *rkey_p = rkey;
    status = UCS_OK;
out_unlock:
    UCP_WORKER_THREAD_CS_EXIT_CONDITIONAL(worker);
    return status;

err_destroy:
    ucp_rkey_destroy(rkey);
    goto out_unlock;
}
```

the same approach we have [here](https://github.com/openucx/ucx/blob/62e93b242236bd93ca1b959a958d30d0d141172a/src/tools/profile/read_profile.c#L141)